src/movegen.cpp

```diff
@@ -443,6 +444,12 @@ namespace {
         // Workaround for passing: Execute a non-move with any piece
         if (pos.pass() && !pos.count<KING>(Us) && pos.pieces(Us))
             *moveList++ = make<SPECIAL>(lsb(pos.pieces(Us)), lsb(pos.pieces(Us)));
+
+        //if "wall or move", generate walling action with null move
+        if (pos.variant()->wallOrMove)
+        {
+            moveList = make_move_and_gating<NORMAL>(pos, moveList, Us, SQ_A1, SQ_A1);
```

Comment:
I think this might potentially be problematic. Currently passes rely on that a piece makes a non-move, but if no piece is placed on a1 this could potentially break here, or at least the `do_move` code has not explicitly been made ready yet for a non-piece non-move.